src/SFML/Window/VideoMode.cpp

```diff
@@ -76,6 +76,11 @@ bool VideoMode::isValid() const
     return std::find(modes.begin(), modes.end(), *this) != modes.end();
 }
 
+////////////////////////////////////////////////////////////
+Vector2u VideoMode::getAspectRatio() const
+{
+    return size / std::gcd(size.x, size.y);
```

Comment:
This should handle the case where gcd is 0. (That happens with a default-constructed VideoMode instance.)

Reply:
You are right